sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -314,12 +316,15 @@ private void setArchitectures(final @NotNull Device device) {
     device.setModelId(Build.ID);
     setArchitectures(device);
 
+    // likely to be applied only if applyScopeData
```

Comment:
the chances that the user opens the app again right away is high, so this won't change much

Reply:
Even though it's more likely that an app that crashes is reopened immediately by the user, I believe we can agree this is also not the case sometimes. And this sometimes multipled by the number of events we'll get through it'll be millions of false data.

That said, we could 'meet in the middle' and verify the timestamp of the event being within "n seconds" of the current time. And if so, apply any of such data.

Similar approach could be used with the performance product (transactions) for data that is costly to retrieve, so that we cache the value for that amount of seconds and hence, accept that the value could be slightly off from when the event actually happened.
How does that sound?

Reply:
I still think the right approach would be doing that thru sentry-native
https://github.com/getsentry/sentry-native/issues/520
hence I'm going to remove free memory and is low memory on this PR, but let's collaborate on the other fields/approach in another PR.